mobile/src/tabbar/tab-navigation-options-factory.tsx

```diff
@@ -1,28 +1,49 @@
 import React from 'react';
 import { NavigationTabScreenOptions } from 'react-navigation';
-import {Platform, Dimensions } from 'react-native';
+import { Dimensions, Platform } from 'react-native';
 import tabBarStyles from './tab-bar-styles';
 import { StyledText } from '../override/styled-text';
 import { ApplicationIcon, ApplicationIconBold } from '../override/application-icon';
 
+//============================================================================
 export class TabNavigationOptionsFactory {
+
+    //----------------------------------------------------------------------------
     public create(label: string, iconName: string): NavigationTabScreenOptions {
+
         return {
             tabBarLabel: this.getTabBarLabel(label),
             tabBarIcon: ({ tintColor, focused }) => {
                 if (focused) {
-                    return <ApplicationIconBold name={iconName} size={Platform.OS === 'ios' ? Dimensions.get('window').width * 0.08 : Dimensions.get('window').width * 0.04} style={tabBarStyles.tabImages} ></ApplicationIconBold>;
+                    return <ApplicationIconBold
+                        name={iconName}
+                        size={this.applicationIconSize()}
+                        style={tabBarStyles.tabImages}/>;
                 } else {
-                    return <ApplicationIcon name={iconName} size={Platform.OS === 'ios' ? Dimensions.get('window').width * 0.08 : Dimensions.get('window').width * 0.04} style={tabBarStyles.tabImages} ></ApplicationIcon>;
+                    return <ApplicationIcon
+                        name={iconName}
+                        size={this.applicationIconSize()}
+                        style={tabBarStyles.tabImages}/>;
                 }
             }
-
         };
     }
 
+    //----------------------------------------------------------------------------
+    // noinspection JSMethodCanBeStatic
+    private applicationIconSize(): number {
+        return Platform.OS === 'ios' ? Dimensions.get('window').width * 0.08 : Dimensions.get('window').width * 0.04;
+    }
+
+    //----------------------------------------------------------------------------
+    // noinspection JSMethodCanBeStatic
     private getTabBarLabel(label: string) {
-        return Platform.OS === 'ios'
-            ? label
-            : <StyledText numberOfLines={1} ellipsizeMode={'tail'} style={tabBarStyles.tabBarLabel}>{label}</StyledText>;  //TODO: fix text width issue on narrow screens
+
+        const labelStyled =
+            <StyledText numberOfLines={1} ellipsizeMode={'tail'} style={tabBarStyles.tabBarLabel}>
```

Comment:
this code repeats here and in tab bar custom. Aware of that?

Reply:
I was aware but didn't actually care. Done. 2d02ece0c6d1053f5dbc0f060d2b8c3350de97f3